samples/matter/common/src/ota_multi_image_downloader.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "ota_multi_image_downloader.h"
+
+#include <dfu/dfu_target.h>
+#include <pm_config.h>
+
+size_t OTAMultiImageDownloaders::sCurrentId;
+
+std::array<OTAMultiImageDownloader, OTAMultiImageDownloaders::CurrentImageID::LIMIT>
+	OTAMultiImageDownloaders::sImageHandlers{
+		{ OTAMultiImageDownloader{ 0, PM_APP_1_CORE_APP_DOWNLOAD_OFFSET, PM_APP_1_CORE_APP_DOWNLOAD_SIZE,
+					   PM_APP_1_CORE_APP_ADDRESS, PM_APP_1_CORE_APP_SIZE },
+		  OTAMultiImageDownloader{ 1, PM_APP_1_CORE_NET_DOWNLOAD_OFFSET, PM_APP_1_CORE_NET_DOWNLOAD_SIZE,
+					   PM_APP_1_CORE_NET_ADDRESS, PM_APP_1_CORE_NET_SIZE },
+		  OTAMultiImageDownloader{ 2, PM_APP_2_CORE_APP_DOWNLOAD_OFFSET, PM_APP_2_CORE_APP_DOWNLOAD_SIZE,
+					   PM_APP_2_CORE_APP_ADDRESS, PM_APP_2_CORE_APP_SIZE },
+		  OTAMultiImageDownloader{ 3, PM_APP_2_CORE_NET_DOWNLOAD_OFFSET, PM_APP_2_CORE_NET_DOWNLOAD_SIZE,
+					   PM_APP_2_CORE_NET_ADDRESS, PM_APP_2_CORE_NET_SIZE } }
+	};
+
+static const device *sExtFlash = DEVICE_DT_GET(DT_CHOSEN(nordic_pm_ext_flash));
+
+int OTAMultiImageDownloader::Init(const device *flash, ssize_t size)
+{
+	if (!device_is_ready(flash)) {
+		ChipLogError(SoftwareUpdate, "Flash device is not ready!");
+		return -EFAULT;
+	}
+	return Init(flash, mOffset, size);
+}
+
+int OTAMultiImageDownloader::Init(const device *flash, size_t offset, size_t size)
+{
+	mBuffer = (uint8_t *)Platform::MemoryAlloc(kBufferSize);
```

Comment:
Shouldn't we check if alloc succeeded before passing potential nullptr further?

Reply:
Yes, thanks for the catch.